pipelines/shopify/incremental_loading_helpers.py

```diff
@@ -0,0 +1,97 @@
+from datetime import datetime, timedelta, timezone
+from dateutil import parser
+from dlt.common.configuration import with_config
+from enum import Enum
+from time import sleep
+from typing import Optional
+
+import dlt
+import re
+
+
+class RelationType(str, Enum):
+    NEXT = 'next'
+    PREVIOUS = 'previous'
+
+
+def get_linked_url(link_header: str, relation_type: str) -> str:
+    """
+    Specific function to extract next/previous linked url from the link header
+    """
+    linked_url = ''
+    if all([isinstance(link_header, str), relation_type in tuple(RelationType)]):
+        link_header_split = link_header.split(', ')
+        link_header_filter = tuple(filter(lambda x: relation_type in x, link_header_split))
+        if len(link_header_filter) == 1:
+            linked_header_match = re.match(r'^.*<(.*)>.*$', link_header_filter[0])
+            if linked_header_match:
+                linked_url = linked_header_match.group(1)
+    return linked_url
+
+
+@with_config
+def get_since_timestamp(endpoint: str, step: int = dlt.config.value, max_retries: int = dlt.config.value, backoff_delay: float = dlt.config.value, days_back: int = dlt.config.value) -> str:
+    """
+    Specific function to generate 'since_timestamp' based on last timestamp stored in dlt's state/destiny (if available)
+    """
+    since_timestamp_str = ''
+    if all([isinstance(step, int), step > 0, isinstance(days_back, int), days_back > 0]):
+        last_timestamp = _get_last_timestamp_from_state(endpoint)
+        if not last_timestamp:
+            last_timestamp = _get_last_timestamp_from_destiny(endpoint, max_retries, backoff_delay)
```

Comment:
why do you need to access destination?

Reply:
Just in case dlt's state gets erased from disk or unavailable for any another reason, and destiny still remains loaded with data.
@adrianbr

Reply:
it should not be necessary. I think this code will not run. I added some docs to state, maybe they help? https://dlthub.com/docs/concepts/state

Reply:
I've removed that block and updated the rest.